pkgs/servers/sql/postgresql/ext/timescaledb-apache.nix

```diff
@@ -0,0 +1,54 @@
+{ lib, stdenv, fetchFromGitHub, cmake, postgresql, openssl, libkrb5 }:
+
+# # To enable on NixOS:
+# config.services.postgresql = let
+#   # The postgresql pkgs has to be taken from the
+#   # postgresql package used, so the extensions
+#   # are built for the correct postgresql version.
+#   postgresqlPackages = config.services.postgresql.package.pkgs;
+# in {
+#   extraPlugins = with postgresqlPackages; [ timescaledb ];
+#   settings.shared_preload_libraries = "timescaledb";
+# }
+
+stdenv.mkDerivation rec {
+  pname = "timescaledb-apache";
+  version = "2.10.0";
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ postgresql openssl libkrb5 ];
+
+  src = fetchFromGitHub {
+    owner = "timescale";
+    repo = "timescaledb";
+    rev = version;
+    sha256 = "sha256-3Pne4L9P8mJv2ja6EpxtpCBCYvlCP6D5CzDtkkvE23c=";
+  };
+
+  cmakeFlags = [ "-DSEND_TELEMETRY_DEFAULT=OFF" "-DREGRESS_CHECKS=OFF" "-DTAP_CHECKS=OFF" "-DAPACHE_ONLY=ON" ]
```

Comment:
The two changes to the unfree version are `pname = "timescaledb-apache";` and `"-DAPACHE_ONLY=ON"` in `cmakeFlags`.

I don't think this requires two separate but almost identical files.
Instead, maybe something like

```diff
diff --git a/pkgs/servers/sql/postgresql/ext/timescaledb.nix b/pkgs/servers/sql/postgresql/ext/timescaledb.nix
index 045e381edd7..1c1fee31c78 100644
--- a/pkgs/servers/sql/postgresql/ext/timescaledb.nix
+++ b/pkgs/servers/sql/postgresql/ext/timescaledb.nix
@@ -1,4 +1,4 @@
-{ lib, stdenv, fetchFromGitHub, cmake, postgresql, openssl, libkrb5 }:
+{ lib, stdenv, fetchFromGitHub, cmake, postgresql, openssl, libkrb5, enableUnfree ? true }:
 
 # # To enable on NixOS:
 # config.services.postgresql = let
@@ -12,7 +12,7 @@
 # }
 
 stdenv.mkDerivation rec {
-  pname = "timescaledb";
+  pname = "timescaledb${lib.optionalString enableUnfree "-apache"}";
   version = "2.10.0";
 
   nativeBuildInputs = [ cmake ];
@@ -26,6 +26,7 @@ stdenv.mkDerivation rec {
   };
 
   cmakeFlags = [ "-DSEND_TELEMETRY_DEFAULT=OFF" "-DREGRESS_CHECKS=OFF" "-DTAP_CHECKS=OFF" ]
+    ++ lib.optionals (!enableUnfree) [ "-DAPACHE_ONLY=ON" ]
     ++ lib.optionals stdenv.isDarwin [ "-DLINTER=OFF" ];
 
   # Fix the install phase which tries to install into the pgsql extension dir,
@@ -48,7 +49,7 @@ stdenv.mkDerivation rec {
     changelog = "https://github.com/timescale/timescaledb/raw/${version}/CHANGELOG.md";
     maintainers = with maintainers; [ marsam ];
     platforms = postgresql.meta.platforms;
-    license = licenses.asl20;
+    license = with licenses; if enableUnfree then tsl else asl20;
     broken = versionOlder postgresql.version "12";
   };
 }
```

and the necessary import in `packages.nix` (you forgot) could pass that `enableUnfree` argument:
```diff
diff --git a/pkgs/servers/sql/postgresql/packages.nix b/pkgs/servers/sql/postgresql/packages.nix
index 2bf9db061a9..a093d97d61f 100644
--- a/pkgs/servers/sql/postgresql/packages.nix
+++ b/pkgs/servers/sql/postgresql/packages.nix
@@ -55,6 +55,7 @@ self: super: {
     temporal_tables = super.callPackage ./ext/temporal_tables.nix { };
 
     timescaledb = super.callPackage ./ext/timescaledb.nix { };
+    timescaledb-apache = super.callPackage ./ext/timescaledb.nix { enableUnfree = false; };
 
     tsearch_extras = super.callPackage ./ext/tsearch_extras.nix { };
```

I don't know if there are specific conventions just for postgres, e.g.  `-` vs `_` in the package name.
Seems to be the first psql extension that provides both an oss and unfree version.

Will be interesting to see how the pinged maintainers decide on this :)

Reply:
Thanks @IndeedNotJames for the suggestions. I have to admit that I don't know anything about nix. I just found it when a community member asked about the nix packages. I'd also argue if TSL is unfree, but that depends on the definition. I guess the licenses for Cassandra or Mongo, but I see those are also marked as unfree though.

For the suggested change, I think the row to define the package name is wrong, or do I misread how this works? I would expect it to add "-apache" when unfree is enabled. At least that's how it reads 😅

Reply:
Oops, yes, my bad^^

```
pname = "timescaledb${lib.optionalString enableUnfree "-apache"}";
```
should be flipped to
```
pname = "timescaledb${lib.optionalString (!enableUnfree) "-apache"}";
```

Fixed my original comment, thank you :)

Reply:
Applied! Thanks for the help! :)